File: src/javascript/ExponentiationRecursive.js
/**
 * @param {number} base 
 * @param {number} exponent 
 */
function exponentiationRecursive(base, exponent) {
    if (exponent <= 1) return base;
    
    return base * exponentiationRecursive(base, exponent -1);
}

function main(){
    // Sample: 3³ = 3*3*3 = 27
    console.log('Exponentiation 3³ => ', exponentiationRecursive(3, 3));
}

main();

File: src/python/exponentiation_recursive.py
""" Recursive exponentiation algorithm """


def exponentiation_recursive(base, exponent):
    """
    Implementation of an exponentiation algorithm using recursion.

    Arguments:
        base (int): Base of operation
        exponent (int): Exponent of operation

    Returns:
        Returns the result of the exponentiation operation.
    """
    if exponent == 0:
        return 1

    return base * exponentiation_recursive(base, exponent - 1)


if __name__ == "__main__":
    print(exponentiation_recursive(5, 2))
    print(exponentiation_recursive(5, 5))
